test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,244 @@
+                    )
+                    .to.changeTokenBalances(iexecPoco, [anyoneA], [depositAmount])
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(AddressZero, anyoneA.address, depositAmount);
+            });
+
+            it('Should deposit native tokens for another address', async () => {
+                const depositForArgs = [anyoneA.address, { value: etherDepositAmount }] as [
+                    string,
+                    { value: BigNumber },
+                ];
+                expect(await iexecPoco.callStatic.depositFor(...depositForArgs)).to.be.true;
+                await expect(iexecPoco.depositFor(...depositForArgs))
+                    .to.changeEtherBalances(
+                        [iexecAdmin, iexecPoco],
+                        [-etherDepositAmount, etherDepositAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [anyoneA], [depositAmount])
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(AddressZero, anyoneA.address, depositAmount);
+            });
+
+            it('Should depositForArray with exact value and good array lengths', async () => {
+                const amounts = [depositAmount.mul(2), depositAmount];
+                const totalAmount = amounts.reduce((a, b) => a.add(b), BigNumber.from(0));
+                const nativeAmountSent = ethers.utils.parseUnits(totalAmount.toString(), 9);
+                const targets = [anyoneA.address, anyoneB.address];
+                const depositForArrayArgs = [amounts, targets, { value: nativeAmountSent }] as [
+                    BigNumber[],
+                    string[],
+                    { value: BigNumber },
+                ];
+
+                expect(await iexecPoco.callStatic.depositForArray(...depositForArrayArgs)).to.be
+                    .true;
+                await expect(iexecPoco.depositForArray(...depositForArrayArgs))
+                    .to.changeEtherBalances(
+                        [iexecAdmin, iexecPoco],
+                        [-nativeAmountSent, nativeAmountSent],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [anyoneA, anyoneB], [...amounts])
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(AddressZero, anyoneA.address, amounts[0])
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(AddressZero, anyoneB.address, amounts[1]);
+            });
+
+            it('Should depositForArray with good array lengths and excess value sent', async () => {
+                const amounts = [depositAmount.mul(2), depositAmount];
+                const totalAmount = amounts.reduce((a, b) => a.add(b), BigNumber.from(0));
+                const excessAmount = depositAmount;
+                const excessAmountWei = ethers.utils.parseUnits(excessAmount.toString(), 9);
```

Comment:
Maybe we should harmonize wording, I see `ether`,`native` & `wei`, what do you think? (Also harmonize as suffix imo "weiBlabla" vs "blablaWei")

Reply:
Done you can check that :)

Reply:
You have typos `deposiNativetTotalAmount`, `deposiNativetFullAmount`,..

Reply:
Thanks !